### PR TITLE
Adding testing infrastructure.

### PR DESCRIPTION
**Pull request by montera82**
### Description
Adding testing infrastructure. Having to move a few things around and refactoring here and there to allow this change in the the least breaking way possiblee.

Still in progress - early feedbacks are welcomed.
https://node-tap.org/docs/using-with/

image

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -25,5 +25,8 @@ module.exports = {
     atomicassets: {
         contract: 'atomicassets',
         collection: 'alien.worlds'
-    }
+    },
+    docs_host: null
 }
+
+export default config;
```
Comment (montera82) [inline config.ts:32]: converted config to an ESM module to allow es6 style  import, which is needed for Tap test to function well.

File: src/app.ts [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-// const path = require('path');
 import path from 'path'
 import fastify, { FastifyInstance, RouteShorthandOptions } from "fastify"
 import fastifyOas from 'fastify-oas';
 import fastifyCors from 'fastify-cors';
 import fastifyAutoload from 'fastify-autoload';
 import fastifyMongo from 'fastify-mongodb';
 import { Server, IncomingMessage, ServerResponse } from "http"
-const config = require('./config')
+import config from '../config';
```
Comment (montera82) [inline src/app.ts:10]: import instead of require now.

File: src/app.ts [hunk 3]
```diff
@@ -51,31 +46,18 @@ server.register(fastifyAutoload, {
...
-
-server.ready().then(async () => {
-    console.log(`Started API server on ${server_address}:${server_port}`);
-    await server.oas();
-}, (err) => {
-    console.error('Error starting API', err)
-});
+server.get('/', async function (request, reply) {
+    return { hello: 'world' }
+  });
```
Comment (montera82) [inline src/app.ts:61]: this is just a fake test route, it won't make it to the final PR

**Comment by montera82** [comment]
gona close this after our discussion with @rkamysz

**Comment by ktamas77** [comment]
> gona close this after our discussion with @rkamysz

is this moved to #7? ready to close?

**Comment by montera82** [comment]
@ktamas77 correct, closing nw as we agreed to focus on #7

**Pull request closed without merging**